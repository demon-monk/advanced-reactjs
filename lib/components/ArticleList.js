import React from 'react';
import Article from './Article';

class ArticleList extends React.PureComponent {

  render() {
    return (
      <div>
        {Object.values(this.props.articles).map((article) =>
          <Article
            article={article}
            key={article.id}
          />
        )}
      </div>
    );
  }
}

export default ArticleList;
