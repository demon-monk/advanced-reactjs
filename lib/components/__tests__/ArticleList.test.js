import React from 'react';
import renderer from 'react-test-renderer';
import ArticleList from '../ArticleList';


describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: 'a',
      b: 'b',
    },
    articleActions: {
      findAuthor: jest.fn(() => ({})), // mock up
    }
  };
  it('renders correctly', () => {
    const tree = renderer.create(
      <ArticleList
        actions={testProps.articleActions}
        articles={testProps.articles}
      />
    ).toJSON();
    // console.log(tree.toJSON());
    // expect(tree.children.length).toBe(2);
    expect(tree).toMatchSnapshot();
  });
});
