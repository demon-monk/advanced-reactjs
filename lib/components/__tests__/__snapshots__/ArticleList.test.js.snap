// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "padding": 50,
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#999",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={undefined}
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={undefined}
    />
  </div>
  <div
    style={
      Object {
        "padding": 50,
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#999",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={undefined}
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={undefined}
    />
  </div>
</div>
`;
